packages/eslint-plugin/src/util/getWrappingFixer.ts

```diff
@@ -27,51 +27,59 @@ interface WrappingFixerParams {
 
 /**
  * Wraps node with some code. Adds parenthesis as necessary.
- * @returns Fixer which adds the specified code and parens if necessary.
+ * @returns Code which adds the specified code and parens if necessary.
  */
-export function getWrappingFixer(
-  params: WrappingFixerParams,
-): TSESLint.ReportFixFunction {
+export function getWrappingCode(params: WrappingFixerParams): string {
```

Comment:
One thing worth noting is that this uses the parent node to infer whether parens are needed (see `if (isWeakPrecedenceParent(node))`). So I'm wondering if that could give mistaken results for a scenario where we're reparenting code?

Reply:
I understand that you are using the above node to infer whether parentheses are needed, but I don't understand this part (So I'm wondering if that could give mistaken results for a scenario where we're reparenting code?). Can you tell me in more detail what the scenario is for reparenting the code?

Reply:
Sorry for taking a bit to get back to you - took me a while to figure out what I meant 😆 . Basically, `getWrappingFixer()` allows you to transform a node whose parent did not change.
```txt
// input
<stuff before> nodeToBeModified <stuff after>
// output
<stuff before> modifiedNode <stuff after>
```
But, _now that getWrappingCode has been extracted_, it can be used instead like 
```txt
// input
<stuff before> nodeToBeModified <stuff after>
// getWrappingCode output
modifiedNode
// used in fixer as
<DIFFERENT stuff before> modifiedNode <DIFFERENT stuff after>
```
In fact, that's the whole reason that you've extracted it in this PR.
```ts
// input
condition ? `${'left' || 'right'}`.trim() : 'lol';
// getWrappingCodeOutput
('left' || 'right')
// used in fixer as 
condition ? ('left' || 'right').trim() : 'lol';
```
So, `getWrappingCode` needs to deduce whether to wrap the expression in parentheses based on the _NEW_ parent, not the existing one.

Used within the `getWrappingFixer` those were the one and the same, so the line `if (isWeakPrecedenceParent(node))` was justified, but _now that getWrappingCode may be used indepdently_, we'd need something like `if (isWeakPrecedenceNewParent(node, newParent))`, in order for the function to give expected results _in general_. 

I'd have to have a longer think to come up with test cases where this discrepancy would manifest, but hopefully this gets you started? If it's still unclear let me know and I can think up some test cases. The bottom line is that we need to be aware of whether parens are needed for the _new_ parent, not the _existing_ parent, in order for `getWrappingCode` to be safe for general use outside of `getWrappingFixer`.

Reply:
this might well be related actually: https://github.com/typescript-eslint/typescript-eslint/pull/8673/files/5a9f63085e1a3b08ae1c63e29deb34ca93b5165d#r1573616588

Reply:
I apologize for not clear question. But the answers were the best!

If my understanding is correct, the parent that the `isWeakPrecedenceNewParent` function checks must be the new parent. In fact, the previous code was checking the parent of the expression, so it was always checking the templete literal node as the parent, so it was not getting normal results.

However, if i slightly modify the code to satisfy the above conditions, i will get the following results depending on the conditions of innerNode and parents condition.

```
// before
true ? `${'test' || ''}`.trim() : undefined;
// after
true ? (('test' || '')).trim() : undefined;
```
In my opinion, it seems better to use parentheses when both the logic of applying parentheses to innerNode and the logic of examining the state of the parent are satisfied. so i revert `getWrappingFixer` and make new `getWrappingCode` function 

I think we're just getting started, so i appreciate your feedback.